prototype/ptolemy-ui/src/grpc.ts

```diff
@@ -0,0 +1,93 @@
+import { GrpcWebFetchTransport } from "@protobuf-ts/grpcweb-transport";
+import { QueryEngineClient } from "./generated/query_engine.client";
+import { RpcOptions, UnaryCall } from "@protobuf-ts/runtime-rpc";
+import { AUTH_TOKEN_KEY } from "./constants";
+import { QueryStatus, QueryStatusResponse } from "./generated/query_engine";
+
+export const grpcClient = () => {
+        const options: RpcOptions = {
+            interceptors: [
+              {
+                // adds auth header to unary requests
+                interceptUnary(next, method, input, options: RpcOptions): UnaryCall {
+                  if (!options.meta) {
+                    options.meta = {};
+                  }
+                      options.meta['Authorization'] = `Bearer ${localStorage.getItem(AUTH_TOKEN_KEY) || ''}`;
+                      options.meta['grpc-accept-encoding'] = 'identity'
+                      options.meta['grpc-encoding'] = 'identity'
+                  return next(method, input, options);
+                }
+              }
+            ],
+        };
+        const transport = new GrpcWebFetchTransport({ baseUrl: import.meta.env.VITE_PTOLEMY_API, ...options})
+        return new QueryEngineClient(transport);
+}
+
+export const runQuery = async (client: QueryEngineClient, query: string) => {
+    const { response: { queryId, success, error } } = await client.query({ query })
+    if (success) {
+        return queryId
+    } else {
+        throw error
+    }
+
+}
+
+const FINAL_STATES = [QueryStatus.CANCELLED, QueryStatus.COMPLETED, QueryStatus.FAILED]
+
+export const waitForQuery = async (client: QueryEngineClient, queryId: string):
+Promise<QueryStatusResponse> => {
+    let { response } = await client.getQueryStatus({ queryId })
+    while (!FINAL_STATES.includes(response.status)) {
+        response = (await client.getQueryStatus({ queryId })).response
+    }
+    return response
+}
+
+export const processData = (data: Uint16Array[]): JSON => {
```

Comment:
damn vscode autocomplete - this should be `Uint8Array[]`
```suggestion
export const processData = (data: Uint8Array[]): JSON => {
```